### ISSUE TITLE
False positive: Dynamic object leak

### ISSUE THREAD

**Issue by stefandrissen** [Riverside-Software/sonar-openedge#1169]
First time user of the ``Dynamic object leak`` rule and finding that it is flagging curious cases:

````
FUNCTION foo LOGICAL PRIVATE ( ) FORWARD. // <-- remove forward declaration and the warning below will disappear

FUNCTION foo RETURN LOGICAL PRIVATE (
):

   DEFINE VARIABLE hq AS HANDLE NO-UNDO.

   CREATE QUERY hq. // Memory allocated to this object is not released in this code block

   RETURN ?.

   FINALLY:
      DELETE OBJECT hq.
   END FINALLY.

END FUNCTION.
````

![Image](https://github.com/user-attachments/assets/a093c2f2-5602-4de3-b79a-c10ba2e5e880)

**Comment by gquerret** [comment]
Confirmed, I will investigate why this issue is reported.

**Comment by gquerret** [comment]
Fixed in develop branch, will be included in next release.
